apollo-ios-pagination/Sources/ApolloPagination/AsyncGraphQLQueryPager.swift

```diff
@@ -0,0 +1,453 @@
+  func loadNext(cachePolicy: CachePolicy) async throws
+  func loadAll(fetchFromInitialPage: Bool) async throws
+  func refetch(cachePolicy: CachePolicy) async
+  func fetch() async
+}
+
+public actor AsyncGraphQLQueryPager<InitialQuery: GraphQLQuery, PaginatedQuery: GraphQLQuery>: AsyncPagerType {
+  private let client: any ApolloClientProtocol
+  private var firstPageWatcher: GraphQLQueryWatcher<InitialQuery>?
+  private var nextPageWatchers: [GraphQLQueryWatcher<PaginatedQuery>] = []
+  let initialQuery: InitialQuery
+  var isLoadingAll: Bool = false
+  var isFetching: Bool = false
+  let nextPageResolver: (PaginationInfo) -> PaginatedQuery?
+  let previousPageResolver: (PaginationInfo) -> PaginatedQuery?
+  let extractPageInfo: (PageExtractionData<InitialQuery, PaginatedQuery>) -> PaginationInfo
+  var nextPageInfo: PaginationInfo? { nextPageTransformation() }
+  var previousPageInfo: PaginationInfo? { previousPageTransformation() }
+
+  var canLoadPages: (next: Bool, previous: Bool) {
+    (canLoadNext, canLoadPrevious)
+  }
+
+  var publishers: (
+    previousPageVarMap: Published<OrderedDictionary<AnyHashable, PaginatedQuery.Data>>.Publisher,
+    initialPageResult: Published<InitialQuery.Data?>.Publisher,
+    nextPageVarMap: Published<OrderedDictionary<AnyHashable, PaginatedQuery.Data>>.Publisher
+  ) {
+    return ($previousPageVarMap, $initialPageResult, $nextPageVarMap)
+  }
+
+  @Published var currentValue: Result<PaginationOutput<InitialQuery, PaginatedQuery>, Error>?
+  private var queuedValue: Result<PaginationOutput<InitialQuery, PaginatedQuery>, Error>?
+
+  @Published var initialPageResult: InitialQuery.Data?
+  var latest: (previous: [PaginatedQuery.Data], initial: InitialQuery.Data, next: [PaginatedQuery.Data])? {
+    guard let initialPageResult else { return nil }
+    return (
+      Array(previousPageVarMap.values).reversed(),
+      initialPageResult,
+      Array(nextPageVarMap.values)
+    )
+  }
+
+  /// Maps each query variable set to latest results from internal watchers.
+  @Published var nextPageVarMap: OrderedDictionary<AnyHashable, PaginatedQuery.Data> = [:]
+  @Published var previousPageVarMap: OrderedDictionary<AnyHashable, PaginatedQuery.Data> = [:]
+  private var tasks: Set<Task<Void, Error>> = []
+  private var taskGroup: ThrowingTaskGroup<Void, Error>?
+  private var watcherCallbackQueue: DispatchQueue
+
+  /// Designated Initializer
+  /// - Parameters:
+  ///   - client: Apollo Client
+  ///   - initialQuery: The initial query that is being watched
+  ///   - extractPageInfo: The `PageInfo` derived from `PageExtractionData`
+  ///   - nextPageResolver: The resolver that can derive the query for loading more. This can be a different query than the `initialQuery`.
+  ///   - onError: The callback when there is an error.
+  public init<P: PaginationInfo>(
+    client: ApolloClientProtocol,
+    initialQuery: InitialQuery,
+    watcherDispatchQueue: DispatchQueue = .global(qos: .background),
```

Comment:
cc: @AnthonyMDev I went back and forth on whether this should be `background` or `default`. Do you have any preferences?

Reply:
I think it should be `.default` personally.

Reply:
Easy fix. Can update it shortly.

Reply:
cc: @AnthonyMDev I've partially reverted this such that the default is `main`, but can be configured by the user. 

I noticed some unusual behaviors that can be triggered on iOS/iPadOS by:
- Having an ongoing request
- Backgrounding the application
- Using a non-`main` callback queue.